packages/core/snapshot/read.go

```diff
@@ -85,68 +78,129 @@ func readSnapshotHeader(reader io.ReadSeeker) (*ledger.SnapshotHeader, error) {
+		}
+
+		sepsBytes := make([]byte, sepsBytesLen)
+		if err := binary.Read(reader, binary.LittleEndian, sepsBytes); err != nil {
+			return nil, fmt.Errorf("unable to read solid entry points: %w", err)
+		}
 
-	if len(data) > 0 {
-		seps = &SolidEntryPoints{}
-		if _, err = serix.DefaultAPI.Decode(context.Background(), data, &seps, serix.WithValidation()); err != nil {
+		ids := make([]tangleold.BlockID, 0)
+		_, err = serix.DefaultAPI.Decode(context.Background(), sepsBytes, &ids, serix.WithValidation())
+		if err != nil {
 			return nil, err
 		}
+		blkIDs = append(blkIDs, ids...)
+		i += len(ids)
 	}
 
+	seps.Seps = blkIDs
+
 	return seps, nil
 }
 
 // readOutputWithMetadata consumes a slice of OutputWithMetadata from the given reader.
-func readOutputWithMetadata(scanner *bufio.Scanner) (outputMetadatas []*ledger.OutputWithMetadata, err error) {
-	scanner.Scan()
-	data := scanner.Bytes()
+func readOutputWithMetadata(reader io.ReadSeeker) (outputMetadatas []*ledger.OutputWithMetadata, err error) {
+	var outputsLen int64
+	if err := binary.Read(reader, binary.LittleEndian, &outputsLen); err != nil {
+		return nil, fmt.Errorf("unable to read outputsWithMetadata bytes len: %w", err)
+	}
 
-	if len(data) > 0 {
-		outputMetadatas = make([]*ledger.OutputWithMetadata, 0)
-		_, err = serix.DefaultAPI.Decode(context.Background(), data, &outputMetadatas, serix.WithValidation())
-		if err != nil {
-			return nil, err
-		}
+	outputsBytes := make([]byte, outputsLen)
+	if err := binary.Read(reader, binary.LittleEndian, outputsBytes); err != nil {
+		return nil, fmt.Errorf("unable to read outputsWithMetadata: %w", err)
+	}
 
-		for _, o := range outputMetadatas {
-			o.SetID(o.M.OutputID)
-			o.Output().SetID(o.M.OutputID)
-		}
+	outputMetadatas = make([]*ledger.OutputWithMetadata, 0)
+	_, err = serix.DefaultAPI.Decode(context.Background(), outputsBytes, &outputMetadatas, serix.WithValidation())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, o := range outputMetadatas {
+		o.SetID(o.M.OutputID)
+		o.Output().SetID(o.M.OutputID)
 	}
 
 	return
 }
 
 // readEpochDiffs consumes a map of EpochDiff from the given reader.
-func readEpochDiffs(scanner *bufio.Scanner) (epochDiffs map[epoch.Index]*ledger.EpochDiff, err error) {
-	epochDiffs = make(map[epoch.Index]*ledger.EpochDiff)
-
-	scanner.Scan()
-	data := scanner.Bytes()
-	if len(data) > 0 {
-		_, err = serix.DefaultAPI.Decode(context.Background(), data, &epochDiffs, serix.WithValidation())
+func readEpochDiffs(reader io.ReadSeeker) (epochDiffs *ledger.EpochDiff, err error) {
+	spent := make([]*ledger.OutputWithMetadata, 0)
+	created := make([]*ledger.OutputWithMetadata, 0)
+
+	// read spent
+	var spentLen int64
+	if err := binary.Read(reader, binary.LittleEndian, &spentLen); err != nil {
+		return nil, fmt.Errorf("unable to read epochDiffs spent len: %w", err)
+	}
+	spentLenInt := int(spentLen)
+	for i := 0; i < spentLenInt; {
+		s, err := readOutputWithMetadata(reader)
```

Comment:
This method reads multiple outputs, but here you use it to consume only one.

Reply:
`readOutputWithMetadata` reads the bytes length of a outputWithMetadata slice, deserialize them and return the slice.